### ISSUE TITLE
Problem when has PIPE operator in value through --define in cli

### ISSUE THREAD

**Issue by marcusedu** [dart-lang/language#3407]
Consider te programm:
```dart
void main(List<String> args){
    print(String.fromEnvironment('SAMPLE', defaultValue: 'error'));
}
```
when run ```dart ./main.dart --define="SAMPLE=success"``` the output is ```success```
but if has **|** (pipe) in the value doesn't works
```dart ./main.dart --define="SAMPLE=success|fails"``` 
the output is ```'fails' is not recognized as an internal command or external command, an operable program, or a batch file.```

dart info output:
```bash
dart info


If providing this information as part of reporting a bug, please review the information
below to ensure it only contains things you're comfortable posting publicly.

#### General info

- Dart 3.1.3 (stable) (Tue Sep 26 14:25:13 2023 +0000) on "windows_x64"
- on windows / "Windows 10 Pro" 10.0 (Build 22621)
- locale is pt-BR

#### Process info

| Memory | CPU | Elapsed time | Command line |
| -----: | --: | -----------: | ------------ |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
```

**Comment by julemand101** [comment]
I suggest moving the issue to the SDK repository (https://github.com/dart-lang/sdk/issues) since this sounds a lot more like a potential Dart implementation issues rather than an issue with the Dart Language Specification.

**Comment by eernstg** [comment]
Right, the language specification documents do not specify how the relevant so-called [compilation environment](https://github.com/dart-lang/language/blob/3cdbe58a7d30e21d9753d201e9cd3ed4da6dab5a/specification/dartLangSpec.tex#L20614) is established, it only says that `String.fromEnvironment` can be used to map a key to a value based on that environment.

I'll transfer this issue to the SDK repository.
